eval.py

```diff
@@ -418,9 +420,51 @@ def get_final_stats(totals: dict[str,Number],
                              totals['total_gt'],
                              totals['total_pred'],
                              totals['total_tightness'])
-    final_stats.update( get_occluded_stats( totals['tp_occluded'],
-                                            totals['total_gt_occluded'],
-                                            final_stats['precision'] ))
+    if task == "singledetrec":
```

Comment:
Do you think it might be helpful to have a one-liner block comment explaining this? I'm not sure this is right, but something like "Calculate competition statistics for the union of the occluded categories of interest." ? Then beneath the else block "Calculate statistics for each individual occluded category". As an external consumer of others' giant chunks of code, I often find these path markers to be quite helpful in reminding me what's going on.